internal/app/user_test.go

```diff
@@ -0,0 +1,204 @@
+)
+
+var (
+	mockName     = "passwall-patron"
+	mockEmail    = "patron@passwall.io"
+	mockPassword = "123456789123456789"
+	mockSecret   = "supersecret"
+	now          = time.Now()
+	uuidN        = uuid.NewV4()
+)
+
+func TestGenerateSchema(t *testing.T) {
+	store, err := initDB()
+	if err != nil {
+		t.Errorf("Error in database initialization ! err:  %v", err)
+	}
+	type args struct {
+		s    storage.Store
+		user *model.User
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{name: "No error ", args: args{
+			s: store,
+			user: &model.User{
+				ID:               0,
+				UUID:             uuid.NewV4(),
+				Name:             mockName,
+				Email:            mockEmail,
+				MasterPassword:   mockPassword,
+				Secret:           mockSecret,
+				Schema:           "users",
+				Role:             "Member",
+				ConfirmationCode: "12345678",
+				EmailVerifiedAt:  time.Time{},
+			},
+		},
+			wantErr: false,
+		},
+		// todo: populate different cases in test table
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := GenerateSchema(tt.args.s, tt.args.user)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GenerateSchema() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			want, err := store.Users().FindByEmail(mockEmail)
+			if err != nil {
+				t.Errorf("FindByEmail() error %v", err)
+			}
+			got.CreatedAt = got.CreatedAt.UTC()
+			got.UpdatedAt = got.UpdatedAt.UTC()
+			got.EmailVerifiedAt = got.EmailVerifiedAt.UTC()
+			if !reflect.DeepEqual(got, want) {
+				t.Errorf("GenerateSchema() got = %v, want %v", got, want)
+			}
+		})
+	}
+	// todo: teardown resources
+}
+
+func TestCreateUser(t *testing.T) {
+	store, err := initDB()
+	if err != nil {
+		t.Errorf("Error in database initialization ! err:  %v", err)
+	}
+
+	type args struct {
+		s       storage.Store
+		userDTO *model.UserDTO
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{"No Error", args{
+			s: store,
+			userDTO: &model.UserDTO{
+				ID:              0,
+				UUID:            uuidN,
+				Name:            mockName,
+				Email:           mockEmail,
+				MasterPassword:  mockPassword,
+				Secret:          mockSecret,
+				Schema:          "users",
+				Role:            "Member",
+				EmailVerifiedAt: now,
+			},
+		},
+			false,
+		},
+		{"Expected Error: No name, email", args{
+			s: store,
+			userDTO: &model.UserDTO{
+				ID:              1,
+				UUID:            uuidN,
+				Name:            "",
+				Email:           "",
+				MasterPassword:  "",
+				Secret:          "",
+				Schema:          "users",
+				Role:            "Member",
+				EmailVerifiedAt: now,
+			},
+		},
+			false, // this should be true and tests should pass in the true statement
+			// however it does not pass, if there is no name, email provided, how it
+			// is possible to create a record on database side. I guess, something is wrong
+			// in creating user
```

Comment:
As I declared here, this test case should return ERR in normal cases, because I am  not able to imagine to record something on database without email and name.